myokit/formats/sbml/_importer.py

```diff
@@ -159,42 +188,101 @@ def model(self, path):
         # Return finished model
         return model
 
-    def _parse_initial_assignments(self, model, comp, refs, node):
+    def _get_namespaces(self, sbml_version: str, log):
```

Comment:
Shouldn't we be getting the namespace from the file, instead of getting it from the version attributes?
Or maybe just check that the file combines the two correctly and complain if it says it's version 4 but uses ns for version 3?

Reply:
I guess that would make sense! I have, however, not found a native function in etree to access the namespaces in a file. What we could do is look at the SBML definitions, where namespaces should be expected to occur in the file, and then look at all those places. But we probably would have to start branching out from the root, because I don't global searches are possible.

What would you prefer?

Reply:
I think you just need to get it from each element separately, so like you're doing with the regex or using `mxml.split()`.
So I could write
```
<model xmlns="namespace">
  <component />
</model>
```
or
```
<sbml:model xmlns:sbml="namespace">
  <sbml:component />
</sbml:model>
```
or
```
<model xmlns="other namespace">
  <component xmlns="namespace" />
</model>
```
and in all of those the `component` tag would be a "{namespace}component".

So no file-wide namespaces, each element has one!

Reply:
I am using split to access the namespace. I would keep the namespace specification in the beginning for now and the maths ns for the mathy bits for now, as otherwise I will have to prepare a comprehensive list of all the namespaces defined within an SBML version and I would rather do that for the next PR, where the different namespaces may have more relevance. At the moment it seems, we are mostly interested in the maths namespace, so we can translate the equations properly and the gobal version ns to find the relevant nodes.